src/nnpack/interface.jl

```diff
@@ -0,0 +1,139 @@
+function check_support(x, k, pad, stride, dilation = 1)
+    fallback = false
+    dilation == 1 || dilation == (1, 1) || (fallback = true)
+    pad_, stride_ = expand(Val{length(k)}, pad), expand(Val{length(k)}, stride)
+    ((size(x, 1) - k[1] + 2 * pad_[1]) % stride_[1] == 0 && (size(x, 2) - k[2] + 2 * pad_[2]) % stride_[2] == 0) || (fallback = true)
+    return pad_, stride_, fallback
+end
+
+function softmax!(x::A) where A<:AbstractVecOrMat{Float64}
+    x = Float32.(x)
+    softmax!(x)
+end
+
+softmax!(x::A) where A<:AbstractVecOrMat{Float32} =
+    nnp_softmax_output(x, x, threadpool = shared_threadpool[])
```

Comment:
These kwargs seem unecessary, since `threadpool` defaults to `shared_threadpool` anyway